### PR TITLE
Save target URL and redirects user after account creation

### PR DESCRIPTION
**Pull request by kcole16**

### CODE REVIEW

File: src/actions/account.js [hunk 1]
```diff
@@ -95,8 +95,8 @@ const wallet = new Wallet()
 
 export const redirectToApp = () => (dispatch, getState) => {
    const state = getState()
-   const nextUrl = (state.account.url && (state.account.url.success_url || state.account.url.public_key)) ? `/login/?${stringify(state.account.url)}` : '/'
```
Comment (vgrichina) [inline src/actions/account.js]: why /login got removed from path? I'm finding hard to follow what's going on.

File: src/components/accounts/CreateAccount.js [hunk 2]
```diff
@@ -4,15 +4,17 @@ import { connect } from 'react-redux'
...
 
 class CreateAccount extends Component {
    state = {
       loader: false,
       accountId: ''
    }
 
-   componentDidMount = () => {}
+   componentDidMount = () => {
+      this.props.handleRefreshUrl(this.props.history.location)
```
Comment (vgrichina) [inline src/components/accounts/CreateAccount.js]: Isn't this already handle automatically in `Routing.js`?

File: src/components/common/PrivateRoute.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ const PrivateRoute = ({component: Component, account, ...rest}) => (
                <Redirect
                   to={{
                      pathname: '/create/',
-                     search: props.location.search,
+                     search: `${props.location.search}${(props.location.search !== '' ? '&' : '?')}redirect_url=${props.location.pathname}`,
```
Comment (vgrichina) [inline src/components/common/PrivateRoute.js]: feel like less confusing would be to use https://www.npmjs.com/package/query-string

something like:

`stringify({...parse(props.location.search), redirect_url: props.location.pathname)`

File: src/actions/account.js (hunk 1 above)
Comment (kcole16) [reply src/actions/account.js]: Generalized this function to redirect to any route, rather than just 'login'. 

The user's initial target url is stored in ```url.redirect_url``` when they're redirected to ```CreateAccount```. 

Should probably rename it from ```redirectToApp``` to something like ```redirectToRoute``` since it's not necessarily an app login.

File: src/components/accounts/CreateAccount.js (hunk 2 above)
Comment (kcole16) [reply src/components/accounts/CreateAccount.js]: Refactored to remove the ```handleRefreshUrl``` call from ```CreateAccount```.
